lib/spack/docs/packaging_guide.rst

```diff
@@ -2789,13 +2798,15 @@ Adding the following to a package:
              "please use a newer release."
     )
```

Comment:
(suggestion) I would recommend emphasizing that `conflicts` can be an arbitrary spec with compiler/version/architecture/dependency portions, and fleshing out this example to demonstrate.

Reply:
I expanded on part of this above and added another couple of examples below that are similar to some I've encountered reviewing PRs (nearly a month ago). Do you want more changes here?